### PR TITLE
fix(DB): The Missing Diplomat Part 11 Fix

### LINKED ISSUE #4871: [Quest]-The-Missing-Diplomat 11/16 (cannot be completed)
##### ISSUE: FACTION SIDE:

Alliance

##### CONTENT PHASE:

20-29

##### SMALL DESCRIPTION:

Quest: [The Missing Diplomat](https://classic.wowhead.com/quest=1249/the-missing-diplomat)
NPC_Start: [Mikhail](https://classic.wowhead.com/npc=4963/mikhail)
NPC_End: [Mikhail](https://classic.wowhead.com/npc=4963/mikhail)
Zone: [Wetlands](https://classic.wowhead.com/wetlands)
Quest ID: 1249

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Retake the quest when its failed and abandoned.

##### CURRENT BEHAVIOUR:

If quest is failed and later abandoned, cant retake it from _Mikhail_. Reset state of NPC not occurs after a while only after worldserver restart.

##### STEPS TO REPRODUCE THE PROBLEM:

Step 1 Go to Mikhail in Wetlands.
Step 2  Fail the quest and abandon
Step 3 No reset state after few mins.

Chromie Issue: https://github.com/chromiecraft/chromiecraft/issues/208

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/a32310772487a7a547b1a239dfaf0967b19f7493

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/97181934-quest-the-missing-diplomat-11-16-cannot-be-completed?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by LannyE**
## Changes Proposed:
-  Make both Mikhail and Slim Active for the duration of event
-  Use hashmap targeting to overcome targeting distance limitations

## Issues Addressed:
- Slim is walking just barely too far to target and send data back to Mikhail on a failed quest so Mikhail never returns to Quest Giver status without a GM respawning Mikhail
- Closes #4871

## Tests Performed:

- I did the how to test changes section instructions.  
- 

## How to Test the Changes:
 
- .add quest 1248
- Complete quest at Mikhail (.quest remove 1249, 1250 before if you already did quests)
- Accept Subdue Slim quest 1249
- Just stand there and wait for a quest fail
- Wait at least 10 seconds more then abandon failed quest.
- Mikhail should allow you to accept the Subdue Slim quest
- Also test by following Slim to his final waypoint when he despawns 
- wait 10 seconds and return to Mikhail and abandon failed quest
- Mikhail should offer the quest again.
- Also test abandoning quest after curbstomping Slim and waiting 10 seconds or so to see if he respawns and mikhail returns to normal.

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested

File: data/sql/updates/pending_db_world/rev_1617170212396609300.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES 
+(4962,0,6,0,40,0,100,0,7,4962,0,0,0,45,1,1,0,0,0,0,10,9432,4963,1,0,0,0,0,0,"Tapoke \"Slim\" Jahn - On Waypoint 7 Reached - Set Data to Mikhail"),
+(4962,0,11,0,38,0,100,0,1,1,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Tapoke \"Slim\" Jahn - On Data Set - Set Active On"),
+(4962,0,12,0,11,0,100,0,0,0,0,0,0,48,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Tapoke \"Slim\" Jahn - On Respawn - Set Active Off"),
+
+(496200,9,8,0,0,0,100,0,1000,1000,0,0,0,45,1,1,0,0,0,0,10,9432,4963,1,0,0,0,0,0,"Tapoke \"Slim\" Jahn - On Script - Set Data to Mikhail");
+
+-- Mikhail
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 4963 AND `id` = 6;
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 496300 AND `id` = 2;
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1617170212396609300.sql:18]: please add `source_type` too when using `DELETE` on `smart_scripts`

Comment (LannyE) [reply data/sql/updates/pending_db_world/rev_1617170212396609300.sql]: Done

**Comment by pklloveyou** [comment]
good job

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by LannyE** [comment]
Is there something left to do on this before we can get it merged?

**Comment by FrancescoBorzi** [comment]
> Is there something left to do on this before we can get it merged?

no, just me forgetting to change the label. Thanks for the reminder ;)

**Pull request merged**